articles/quickstart/native/ios-swift/09-sign-in-with-apple.md

```diff
@@ -0,0 +1,197 @@
+---
+
+<!-- markdownlint-disable MD002 MD041 -->
+
+## Before You Start
+
+Before you continue with this tutorial, make sure that you have completed the previous tutorials. This tutorial assumes that:
+* You have integrated [Auth0.swift](https://github.com/auth0/Auth0.swift/) as a dependency in your project. 
+* You are familiar with presenting the login screen. To learn more, see the [Login](/quickstart/native/ios-swift/00-login) tutorial.
+* You have an [Apple Developer](https://developer.apple.com/programs/) account, which is a paid account with Apple. (There is no free trial available unless you are part of their [iOS Developer University Program](https://developer.apple.com/support/compare-memberships/)).
+* You have configured your Auth0 tenant and application client to enable **Sign In With Apple** to work. See [Add Sign In with Apple to Your Native App](/articles/connections/references/apple-native/guides/add-siwa-to-native-app) for details on how to do this.
+
+## Add the "Sign In With Apple" Button
+
+As an example of adding the sign-in button to your view, add a `UIStackView` to your main storyboard and create an outlet in `ViewController.swift`:
+
+```swift
+ @IBOutlet weak var loginProviderStackView: UIStackView!
+```
+
+Next, add the `AuthenticationServices` Framework to your `ViewController.swift`:
+
+```swift
+import AuthenticationServices
+```
+
+Add a function that attaches a `ASAuthorizationAppleIDButton` button to the stack view:
+
+```swift
+func setupProviderLoginView() {
+  // Create Button
+  let authorizationButton = ASAuthorizationAppleIDButton()
+
+  // Add Callback on Touch
+  authorizationButton.addTarget(self, action: #selector(handleAuthorizationAppleIDButtonPress), for: .touchUpInside)
+
+  //Add button to the UIStackView
+  self.loginProviderStackView.addArrangedSubview(authorizationButton)
+}
+```
+
+Call this function in `viewDidLoad`:
+
+```swift
+override func viewDidLoad() {
+    super.viewDidLoad()
+
+    setupProviderLoginView()
+}
+```
+
+## Handle Authorization
+
+The following code shows the implementation of a button handler that initializes an authorization request and launches
+the native SIWA flow:
+
+```swift
+@objc
+func handleAuthorizationAppleIDButtonPress() {
+    // Create the authorization request
+    let request = ASAuthorizationAppleIDProvider().createRequest()
+
+    // Set scopes
+    request.requestedScopes = [.email, .fullName]
+
+    // Setup a controller to display the authorization flow
+    let controller = ASAuthorizationController(authorizationRequests: [request])
+
+    // Set delegates to handle the flow response.
+    controller.delegate = self
+    controller.presentationContextProvider = self
+
+    // Action
+    controller.performRequests()
+}
+```
+
+:::note
+At this time the only supported scopes are `email` and `fullname`. For more information, see [ASAuthorization.scope](https://developer.apple.com/documentation/authenticationservices/asauthorization/scope).
```

Comment:
Maybe not the only scopes Apple support?

Reply:
Verified that the `fullName` scope does appear to work after all, at least in the simulator.